nsv13/code/modules/munitions/munitions.dm

```diff
@@ -403,6 +428,106 @@
+	. = FALSE
+	if(loaded && maint_state == 0)
+		to_chat(user, "<span class='warning'>You cannot open the maintence panel while [src] is loaded!</span>")
+		return TRUE
+	else if(!loaded && maint_state == 0)
+		to_chat(user, "<span class='notice'>You begin unfastening the maintenance panel on [src]...</span>")
+		if(tool.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'> You unfasten the maintenance panel on [src].</span>")
+			maint_state = 1
+			update_overlay()
+			return TRUE
+	else if(maint_state == 1)
+		to_chat(user, "<span class='notice'>You begin fastening the maintenance panel on [src]...</span>")
+		if(tool.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'> You fasten the maintenance panel on [src].</span>")
+			maint_state = 0
+			update_overlay()
+			return TRUE
+
+/obj/structure/ship_weapon/wrench_act(mob/user, obj/item/tool)
+	. = FALSE
+	if(maint_state == 1)
+		to_chat(user, "<span class='notice'>You begin unfastening the inner casing bolts on [src]...</span>")
+		if(tool.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>You unfasten the inner case bolts on [src].</span>")
+			maint_state = 2
+			update_overlay()
+			return TRUE
+	else if(maint_state == 2)
+		to_chat(user, "<span class='notice'>You begin fastening the inner casing bolts on [src]...</span>")
+		if(tool.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>You fasten the inner case bolts on [src].</span>")
+			maint_state = 1
+			update_overlay()
+			return TRUE
+
+/obj/structure/ship_weapon/crowbar_act(mob/user, obj/item/tool)
+	. = FALSE
+	if(maint_state == 2)
+		to_chat(user, "<span class='notice'>You begin prying the inner casing off [src]...</span>")
+		if(tool.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>You pry the inner casing off [src].</span>")
+			maint_state = 3
+			update_overlay()
+			if(prob(50))
+				to_chat(user, "<span class='warning'>Oil spurts out of the exposed machinery!</span>")
+				new /obj/effect/decal/cleanable/oil(user.loc)
+				reagents.clear_reagents()
+			return TRUE
+	if(maint_state == 3)
+		to_chat(user, "<span class='notice'>You begin slotting the inner casing back in [src]...</span>")
+		if(tool.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>You slot the inner casing back in [src].</span>")
+			maint_state = 2
+			update_overlay()
+			return TRUE
+
+/obj/structure/ship_weapon/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/reagent_containers))
+		if(maint_state != 3)
+			to_chat(user, "<span class='notice'>You require access to the inner workings of [src].</span>")
+			return
+		else if(maint_state == 3)
+			if(I.reagents.has_reagent(/datum/reagent/oil, 10))
+				to_chat(user, "<span class='notice'>You start lubricating the inner workings of [src]...</span>")
+				do_after(user, 2 SECONDS, target=src)
```

Comment:
This isnt how do after works.

Reply:
So do_after doesn't cause a little potentially interruptible delay with a progress bar?

Reply:
You're looking for:
if(do_after(user, target=src, time=2 SECONDS)
>>After completion

Reply:
No you do an early return not a nest

Reply:
So what do you want to use?

Reply:
If(! doafter(whatever))
    Return

Reply:
Its already done.

Reply:
resolved: incorporated changes